src/main/java/org/opensearch/security/securityconf/ConfigModelV7.java

```diff
@@ -147,6 +148,9 @@ private ActionGroupResolver reloadActionGroups(SecurityDynamicConfiguration<Acti
                 if(actionGroupAsObject != null && actionGroupAsObject instanceof List) {
                     
                     for (final String perm: ((List<String>) actionGroupAsObject)) {
+                        if (perm == entry) {
```

Comment:
Not a proper way to compare `Strings` in Java.
Not a proper way to identify recursion except for a self reference trivial case.